pkg/api/alerting.go

```diff
@@ -296,6 +301,11 @@ func UpdateAlertNotificationByUID(c *models.ReqContext, cmd models.UpdateAlertNo
 	cmd.OrgId = c.OrgId
 	cmd.Uid = c.Params("uid")
 
+	err := fillWithSecureSettingsDataByUID(&cmd)
+	if err != nil {
+		return Error(500, "Failed to update alert notification", err)
+	}
+
```

Comment:
```suggestion

	// remove unencrypted settings
	if url, ok := cmd.SecureSettings["url"]; ok && url != "" {
		cmd.Settings.Del("url")
	}
	if token, ok := cmd.SecureSettings["token"]; ok && token != "" {
		cmd.Settings.Del("token")
	}

```

Reply:
Same as https://github.com/grafana/grafana/pull/25114/files#r445964542

Reply:
Since this only applies to slack I would like to wait with this.